CMakeLists.txt

```diff
@@ -57,11 +57,27 @@ else()
   set(MENDER_USE_NLOHMANN_JSON 0)
 endif()
 
-add_library(mender_compiler_flags INTERFACE)
-target_compile_features(mender_compiler_flags INTERFACE cxx_std_11)
+#---------------------------------------------------------------------------------------------------
+# When deciding which one(s) of these to include, the general rule is this:
+#
+# If it's a cross platform component, add this:
+#   target_link_libraries(<TARGET> PUBLIC crossplatform_compiler_flags)
+#
+# If it's a platform specific component, with a cross platform interface, add this:
+#   target_link_libraries(<TARGET> PRIVATE platform_compiler_flags)
+#   target_link_libraries(<TARGET> INTERFACE crossplatform_compiler_flags)
+
+add_library(crossplatform_compiler_flags INTERFACE)
```

Comment:
See [this reply](https://github.com/mendersoftware/mender/pull/1132#discussion_r1088634549).

Reply:
It seems they have this also: `set_target_properties(myTarget PROPERTIES CXX_EXTENSIONS OFF)` ?

Reply:
That flag is about "upgrading" from std to GNU extensions, which is a different thing. It probably should be off for us actually (the default is on), but since with this code we are adding an explicit standards flag anyway, it probably does not matter.